src/tm-threads.c

```diff
@@ -1736,6 +1736,26 @@ TmEcode TmThreadSpawn(ThreadVars *tv)
         return TM_ECODE_FAILED;
     }
 
+    if (threading_set_stack_size) {
+        /* Adjust thread stack size if configured */
+        SCLogDebug("Setting per-thread stack size to %" PRIu64, threading_set_stack_size);
+        rc = pthread_attr_setstacksize(&attr, (size_t)threading_set_stack_size);
+        if (rc) {
+            printf("WARNING; unable to increase stack size to %" PRIu64 "\n",
```

Comment:
SCLogError? or even FatalError?

Should we first check the value and only set it if is lower than the threshold? This would make it a `minimum-stack-size` I suppose.

Reply:
I'll change `printf` to `FatalError` since this condition is terminal. I'll also update, in a separate commit, the other usage of `printf` in this function.

Regarding `stack-size` (current) vs `minimum-stack-size`. I think either would make sense but since the default stack size requires adjustment on very few platforms, I opted for `stack-size`.